### PR TITLE
Compatible for curator5.

### PR DESCRIPTION
**Pull request by horizonzy**
## What is the purpose of the change
Compatible with curator5.

### CODE REVIEW

File: dubbo-remoting/pom.xml [hunk 1]
```diff
@@ -35,6 +35,7 @@
         <module>dubbo-remoting-http</module>
         <module>dubbo-remoting-netty</module>
         <module>dubbo-remoting-zookeeper</module>
+        <module>dubbo-remoting-zookeeper-curator5</module>
```
Comment (AlbumenJ) [inline dubbo-remoting/pom.xml:38]: Please add module shading support in dubbo-distribution.

Comment (horizonzy) [reply dubbo-remoting/pom.xml:38]: Done.

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Comment by AlbumenJ** [comment]
@kylixs PTAL

File: dubbo-configcenter/dubbo-configcenter-zookeeper/src/main/java/org/apache/dubbo/configcenter/support/zookeeper/ZookeeperDynamicConfigurationFactory.java [hunk 2]
```diff
@@ -28,6 +29,12 @@
 
     private ZookeeperTransporter zookeeperTransporter;
 
+    {
+        ZookeeperTransporter extension = ZookeeperTransporter.getExtension();
```
Comment (kylixs) [inline dubbo-configcenter/dubbo-configcenter-zookeeper/src/main/java/org/apache/dubbo/configcenter/support/zookeeper/ZookeeperDynamicConfigurationFactory.java:33]: It's better to put it into the constructor.

File: dubbo-metadata/dubbo-metadata-report-zookeeper/src/main/java/org/apache/dubbo/metadata/store/zookeeper/ZookeeperMetadataReportFactory.java [hunk 3]
```diff
@@ -28,6 +29,12 @@
 
     private ZookeeperTransporter zookeeperTransporter;
 
+    {
+        ZookeeperTransporter extension = ZookeeperTransporter.getExtension();
```
Comment (kylixs) [inline dubbo-metadata/dubbo-metadata-report-zookeeper/src/main/java/org/apache/dubbo/metadata/store/zookeeper/ZookeeperMetadataReportFactory.java:33]: It's better to put it into the constructor.

File: dubbo-registry/dubbo-registry-zookeeper/src/main/java/org/apache/dubbo/registry/zookeeper/ZookeeperRegistryFactory.java [hunk 4]
```diff
@@ -17,22 +17,24 @@
...
 public class ZookeeperRegistryFactory extends AbstractRegistryFactory {
 
     private ZookeeperTransporter zookeeperTransporter;
 
-    /**
-     * Invisible injection of zookeeper client via IOC/SPI
-     * @param zookeeperTransporter
-     */
+    {
+        ZookeeperTransporter extension = ZookeeperTransporter.getExtension();
```
Comment (kylixs) [inline dubbo-registry/dubbo-registry-zookeeper/src/main/java/org/apache/dubbo/registry/zookeeper/ZookeeperRegistryFactory.java:33]: It's better to put it into the constructor.

File: dubbo-configcenter/dubbo-configcenter-zookeeper/src/main/java/org/apache/dubbo/configcenter/support/zookeeper/ZookeeperDynamicConfigurationFactory.java (hunk 2 above)
Comment (horizonzy) [reply dubbo-configcenter/dubbo-configcenter-zookeeper/src/main/java/org/apache/dubbo/configcenter/support/zookeeper/ZookeeperDynamicConfigurationFactory.java:33]: fine.

**Pull request merged**